convokit/model/corpus.py

```diff
@@ -33,181 +46,334 @@ class Corpus:
+                    if self.storage.raw_version != self.storage.version:
+                        if utterances is None:
+                            tmp_storage = StorageManager(
+                                storage_type,
+                                db_host=db_host,
+                                data_directory=data_directory,
+                                corpus_id=self.id,
+                                version=self.storage.raw_version,
+                                in_place=True,
+                            )
+                            tmp_storage.setup_collections(
+                                Utterance, Conversation, Speaker, ConvoKitMeta
+                            )
+
+                            self.copy_from_storage(tmp_storage)
+                            print(
+                                f"Copying corpus {self.id}_v{self.storage.raw_version} "
+                                f"to corpus {self.storage.full_name}"
+                            )
+
+                        else:
+                            warn(
+                                f"Corpus {self.id}_v{self.storage.raw_version} found "
+                                "in the DB but utterances != None; ignoring existing data and using given utterances"
+                            )
+                    else:
+                        print(
+                            f"Corpus {self.storage.full_name} not found in the DB; building new corpus, initially was not in place"
+                        )
+                else:
+                    assert self.storage.raw_version == self.storage.version
+                    if self.id in self.storage.db.list_collections():
+                        # if any([self.id in k for k in [x["name"] for x in list(self.storage.db.list_collections())]]):
+                        print(f"Connecting to corpus {self.storage.full_name} in place in the DB")
+                    else:
+                        print(
+                            f"Corpus {self.storage.full_name} not found in the DB; building new corpus, in place."
+                        )
+
+            elif self.storage.storage_type == "mem":
+                convos_data = defaultdict(dict)
+                if exclude_utterance_meta is None:
+                    exclude_utterance_meta = []
+                if exclude_conversation_meta is None:
+                    exclude_conversation_meta = []
+                if exclude_speaker_meta is None:
+                    exclude_speaker_meta = []
+                if exclude_overall_meta is None:
+                    exclude_overall_meta = []
+
+                # Construct corpus from file or directory
+                if self.storage.corpus_id is not None or filename is not None:
+                    if filename is None:
+                        filename = os.path.join(self.storage.data_directory, self.id)
+                    print(f"Loading corpus {self.id} from disk at {filename}")
```

Comment:
In the documentation, we explicitly state that the `filename` parameter is for use in mem mode and the `corpus_id` parameter is for use in DB mode. To me, this implies that supplying `corpus_id` in mem mode should either get ignored or throw an error, and likewise for supplying `filename` in DB mode. But here we see that this is not what happens; instead in mem mode `corpus_id` gets treated as a sort of backup for `filename`, with the code automatically inferring a filename from `corpus_id` if `filename` is not provided.

Either the code or the documentation needs to change. If we think the current behavior is acceptable, we need to clarify in the documentation that `corpus_id` is not exclusive to DB mode. If on the other hand we want to strictly stick the what the documentation currently implies, then this code needs to go *and* we need to add a warning or error when `filename` and `corpus_id` are used with the wrong storage mode.